spring-test/src/main/java/org/springframework/test/web/client/match/MockRestRequestMatchers.java

```diff
@@ -79,11 +79,13 @@ public static RequestMatcher requestTo(final Matcher<String> matcher) {
 	/**
 	 * Assert the request URI string.
 	 * @param expectedUri the expected URI
+	 * @param uriVars zero or more URI variables to populate the expected URI
 	 * @return the request matcher
 	 */
-	public static RequestMatcher requestTo(final String expectedUri) {
+	public static RequestMatcher requestTo(final String expectedUri, final Object... uriVars) {
 		Assert.notNull(expectedUri, "'uri' must not be null");
-		return request -> assertEquals("Request URI", expectedUri, request.getURI().toString());
+		String uri = UriComponentsBuilder.fromUriString(expectedUri).buildAndExpand(uriVars).encode().toString();
```

Comment:
In regards to the encoded expectedUri, I looked at the recent changes to use URI vars in [MockMvcResultMatchers](https://github.com/spring-projects/spring-framework/blob/master/spring-test/src/main/java/org/springframework/test/web/servlet/result/MockMvcResultMatchers.java#L90) and modeled this chage off that.

Which option do you prefer? The first option is certainly more convenient. The separate method option also works well because if you are going back to refactor `requestTo` to use URI vars, then you can change the method name while at it.